packages/blade/src/components/SideNav/_decisions/decisions.md

```diff
@@ -0,0 +1,545 @@
+
+1. All items should be accessible by `TAB`. Including going between levels L1, L2, L3
+2. We should use Blade's `SkipNav` utility to provide option of skipping nav and going to content
+3. React Router automatically handles `aria-current="page"`. Verify that it is working as expected.
+
+## References
+
+- [RazorpayX Navigation Bar](https://github.com/razorpay/x/blob/master/src/js/views/Home/ActionCenter/index.js#L21) (Internal)
+- [Razorpay Merchant Dashboard Navigation Bar](https://github.com/razorpay/dashboard/blob/6f61a5909c5311c3be08a3ea28b920ed302f1e64/web/js/merchant/components/SidebarV2/Sidebar.tsx) (Internal)
+- [SideNavigation - Atlassian DS](https://atlassian.design/components/side-navigation/examples)
+- [SideNav - Primer / GitHub](https://primer.style/react/deprecated/SideNav#url)
+- [SideNav - Carbon DS](https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--fixed-side-nav)
+- [SideNav - BaseWeb DS / Uber](https://baseweb.design/components/side-nav/#colors)
+- [sidenav - Spectrum / Adobe](https://opensource.adobe.com/spectrum-web-components/components/sidenav/)
+
+## Open Questions
+
+### Dev
+
+- #### Parent Component Name: `SideNav` vs `Sidebar` vs `Navbar` vs `Nav`
+
+  SideNav
+
+- #### Navigation Item Naming Decision: `SideNavLink` vs `SideNavItem` vs `SideNavItemLink`
+
+  SideNavLink
+
+- #### `<SideNav routerLink={NavLink} />` vs `<SideNavLink as={NavLink} />`
+
+  1. We pass react router on parent SideNav component
+
+  - Example
+    ```jsx
+    <SideNav routerLink={NavLink}>
+      <SideNavLink />
+      <SideNavLink />
+    </SideNav>
+    ```
+  - Pros: Router Link defined at one place on top so less likely to be missed / changed between multiple items
+  - Cons: Uncommon pattern
+
+  2. We pass it from as prop on SideNavLink component
+
+  - Example
+    ```jsx
+    <SideNav>
+      <SideNavLink as={NavLink} />
+      <SideNavLink as={NavLink} />
+    </SideNav>
+    ```
+  - Pros: Common pattern
+  - Cons: Requires defining NavLink at each item (for projects not part of One Dashboard like Admin Dashboard, Bank Portals, etc)
+
+  ##### Conclusion
+
+  We are going with `as={NavLink}`
+
+- #### Why `isActive` prop? Why not handle active state internally
+
+  Earlier we started the API by saying that marking of the link as active based on route will be handled internally in blade component. There are some implemetation constraints when we go with this approach of handling active link internally
+
+  1. We don't have access to hooks like `useLocation` or utilities like `matchPath` inside Blade since blade is library indepedent of the routing logic
+  2. Although React Router's NavLink automatically adds `aria-current="page"` to active link which can be used to change colors in CSS, it doesn't give us access to handling state internally for our L1 -> L2 navigations [Here's the related proposal I created in React Router's Repo](https://github.com/remix-run/react-router/discussions/11543).
```

Comment:
NavLink internally maintains the isActive based on route defined right? Can we not make use of that?


> it doesn't give us access to handling state internally for our L1 -> L2 navigations

can you please elaborate on this a little?

Reply:
> NavLink internally maintains the isActive based on route defined right? Can we not make use of that?

It doesn't give any `isActive` in JS / React state. It adds aria-current="page" to link which can be used for CSS styling but because we have multiple menus where lets say you want to open L2 from L1, you need a way to know which item is Active in the react state (Added a bit more context in that React Router's proposal)

> NavLink internally maintains the isActive based on route defined right

Basically the issue is its "only" internally maintaining that active state inside React Router's NavLink. Its not giving us a way to access that active state in our component's state